### DISCUSSION TITLE
"orchestration function not found when running multiple func hosts" on localhost

### DISCUSSION THREAD

**Discussion by mizrael** [Azure/azure-functions-durable-extension#2175] [Q&A] [answered]
I'm trying to understand if this is by design or if there's any special config setting that I need. Here's my scenario:
I'm developing and running a bunch of different Azure Function projects, each one using Durable Functions.
For various reasons they're not part of the same solution, so I'm forced to run them via different VS instances.
Now, I've noticed that when at least one Orchestration is running, it often happens that one of the other Hosts throws an error like this:

```
The function 'XXXXXXXXX' doesn't exist, is disabled, or is not an orchestrator function. Additional info: The following are the known orchestrator functions: ................. '.. InstanceId: a6496743-83f1-4228-86ab-dcbaf916bbe5. Function: XXXXXXXXX. HubName: TestHubName. AppName: . SlotName: . ExtensionVersion: 2.2.1. SequenceNumber: 165.
```

And of course, the Host throwing is not the one that is supposed to contain the expected Function.

I guess that the Func Hosts are all sharing the same queues on my local Storage Emulator, which is of course causing a "competing consumers" issue.

Is there a solution to this?

**Comment by cgillum** [comment, marked as answer]
Yeah, it sounds like your two function apps are sharing the same task hub and the same storage account (i.e. the same storage emulator instance). The simple work-around is to configure each of your function apps with a different task hub name in host.json. More info here: https://docs.microsoft.com/en-us/azure/azure-functions/durable/durable-functions-task-hubs?tabs=csharp

```json
{
  "version": "2.0",
  "extensions": {
    "durableTask": {
      "hubName": "TaskHubA"
    }
  }
}
```

```json
{
  "version": "2.0",
  "extensions": {
    "durableTask": {
      "hubName": "TaskHubB"
    }
  }
}
```

**Reply by mizrael** [reply]
awesome! I was really hoping there was a simple solution like this, thanks!
I was battling between spinning up multiple Azurite instances as containers or creating new Storage Accounts on Azure :D
